protocol/binary_test.go

```diff
@@ -776,6 +791,24 @@ func TestBinaryEnvelopeSuccessful(t *testing.T) {
 			continue
 		}
 
+		// Also verify whether we can infer the presence of the envelope
```

Comment:
Same thing - wondering if the envelope agnostic items should be in their own test.